src/main/java/seedu/address/model/event/RecurringEvent.java

```diff
@@ -25,7 +94,42 @@ public int compareTo(RecurringEvent o) { //TODO: Add implementation for compareT
      */
     @Override
     public boolean occursBetween(LocalDateTime startPeriod, LocalDateTime endPeriod) {
-        // TODO: Implement this method.
-        return false;
+
+        DayOfWeek startPeriodDay = startPeriod.getDayOfWeek();
+        DayOfWeek endPeriodDay = endPeriod.getDayOfWeek();
+
+        LocalTime startPeriodTime = LocalTime.parse(startPeriod.format(DateTimeFormatter.ofPattern("HH:mm")));
+        LocalTime endPeriodTime = LocalTime.parse(startPeriod.format(DateTimeFormatter.ofPattern("HH:mm")));
+
+        boolean isDayBetweenPeriod = false;
+        boolean isTimeBetweenPeriod;
+
+        long daysBetween = startPeriod.until(endPeriod, ChronoUnit.DAYS);
+
+        if (daysBetween > 7) {
+            return true;
+        }
+
+        DayOfWeek eventDay = startPeriodDay;
+        for (int i = 0; i < daysBetween; i++) {
+            if (this.dayOfWeek.equals(eventDay)) {
+                isDayBetweenPeriod = true;
+                break;
+            }
+            eventDay = eventDay.plus(1);
+        }
+
+        if (eventDay.equals(startPeriodDay) && eventDay.equals(endPeriodDay)) {
+            isTimeBetweenPeriod = (this.startTime.isAfter(startPeriodTime) || this.startTime.equals(startPeriodDay))
+                    && (this.endTime.isBefore(endPeriodTime) || this.endTime.equals(endPeriodTime));
+        } else if (eventDay.equals(startPeriodDay)) {
+            isTimeBetweenPeriod = this.startTime.isAfter(startPeriodTime) || this.startTime.equals(startPeriodDay);
+        } else if (eventDay.equals(endPeriodDay)) {
+            isTimeBetweenPeriod = this.endTime.isBefore(endPeriodTime) || this.endTime.equals(endPeriodTime);
+        } else {
+            isTimeBetweenPeriod = true;
+        }
+
+        return isDayBetweenPeriod && isTimeBetweenPeriod;
```

Comment:
This implementation seems quite complex. Perhaps you can consider the cases in which the Event can occur between the period:
1. If the start/end of the event occurs between the period
2. Start before/equal to start of period && end equal/after end of period